reference/multipart.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: page
+title: Multipart Encoding
+---
+
+Praxis has built-in support for handling "multipart/form-data" (link to RFC?) encoded requests and responses, in the form of the `Praxis::Types::MultipartArray` type, and the `Praxis::Responses::MultipartOk` response.
+
+
+## Praxis::Types::MultipartArray
+
+The `MultipartArray` type is an `Attributor::Collection` of `Praxis::MultipartPart` members that allows you to specify the handling of each part of a multipart request or response body.
+
+### Definition
```

Comment:
Describing the body type of a `MultipartArray` consists of defining shape of it parts. There are three ways to define the parts, ranging from being very loose, to very specific:
* One can loosely define all the parts have the same payload and name type (Which could be set to `Object` to imply no restrictions or coercions to be applied).
* One can also define the same expected payload type (and headers) for a group of parts with names matching a regular expression.
* Or one can define the payload of a type (and headers) for a specific type by name.

Here is an example of each of these cases:
TODO 3 examples.

Parts, can also be defined to contain files (and match a filename). To do that you can pass the `filename: true` option when defining the part (or equivalently use the `file` method instead for syntactic sugar). Use the `filename` DSL inside the part definition to express any expected string or pattern for it.

Here are some examples for defining file parts:
TODO ...